p2pp/gcode.py

```diff
@@ -97,9 +97,8 @@ def __str__(self):
         for key in self.Parameters:
             if self.fullcommand not in ['G0', 'G1'] or key not in sorted_keys:
                 value = self.Parameters[key]
-                if not value:
+                if not isinstance(value, (int, float)) and not value:
```

Comment:
@tomvandeneede Need to check type here or

"G92 E0.0" is reformatted as "G92 E\n"

Added a test case for this as well.